budgetkey_data_pipelines/pipelines/procurement/tenders/exemptions/parse_exemption_data.py

```diff
@@ -0,0 +1,108 @@
+    def __init__(self, **kwargs):
+        self._base_url = kwargs.pop("base_url", BASE_URL)
+        super(ParseExemptionDataProcessor, self).__init__(default_input_resource=INPUT_RESOURCE,
+                                                          default_output_resource=OUTPUT_RESOURCE,
+                                                          default_replace_resource=True,
+                                                          table_schema=TABLE_SCHEMA,
+                                                          **kwargs)
+
+    def filter_resource_data(self, data, parameters):
+        for exemption in data:
+            page = pq(exemption["data"])
+            documents = []
+            for update_time_elt, link_elt, img_elt in zip(page("#ctl00_PlaceHolderMain_pnl_Files .DLFUpdateDate"),
+                                                          page("#ctl00_PlaceHolderMain_pnl_Files .MrDLFFileData a"),
+                                                          page("#ctl00_PlaceHolderMain_pnl_Files .MrDLFFileData img")):
+                documents.append({"description": img_elt.attrib.get("alt", ""),
+                                  "link": "{}{}".format(BASE_URL, link_elt.attrib.get("href", "")),
+                                  "update_time": update_time_elt.text,})
+            source_data = {
+                k: page("#ctl00_PlaceHolderMain_lbl_{}".format(v)).text() for k, v in INPUT_FIELDS_TEXT_MAP.items()}
+            yield {
+                "publisher_id": int(exemption["pid"]),
+                "publication_id": int(source_data["publication_id"]),
+                "page_url": exemption["url"],
+                "description": source_data["description"],
+                "supplier_id": int(source_data["supplier_id"]),
+                "supplier": source_data["supplier"],
+                "contact": source_data["contact"],
+                "publisher": source_data["publisher"],
+                "contact_email": source_data["contact_email"],
+                "claim_date": datetime.strptime(source_data["claim_date"], "%d/%m/%Y").date() if source_data["claim_date"] else None,
+                "last_update_date": datetime.strptime(source_data["last_update_date"], "%d/%m/%Y").date() if source_data["last_update_date"] else None,
+                "reason": source_data["reason"],
+                "source_currency": source_data["source_currency"],
+                "regulation": source_data["regulation"],
+                "volume": source_data["volume"],
+                "subjects": source_data["subjects"],
+                "start_date": datetime.strptime(source_data["start_date"], "%d/%m/%Y").date() if source_data["start_date"] else None,
+                "end_date": datetime.strptime(source_data["end_date"], "%d/%m/%Y").date() if source_data["end_date"] else None,
+                "decision": source_data["decision"],
+                "page_title": source_data["page_title"],
+                "documents_json": json.dumps(documents)
```

Comment:
changed, but it means we now require to use postgresql DB (sqlite won't work for array / object types)
so I guess README will need to be updated as quickstart assumes sqlite DB